### ISSUE TITLE
Hangs indefinitely on inet_gethost 4 since 26.2.5.4

### ISSUE THREAD

**Issue by rafaelgaspar** [erlang/otp#8961]
**Describe the bug**

The application randomly hangs indefinitely on inet_gethost 4.

**To Reproduce**

I don't have a minimal example of this happening unfortunately, and it's intermittent. See additional context.

**Expected behavior**

To not hang indefinitely.

**Affected versions**

Started since we updated to 26.2.5.4.

**Additional context**

Since the update to 26.2.5.4 we noticed that our application started hanging indefinitely while starting the application.

Checking the process tree it seams to get stuck always similar to this one:

![Screenshot 2024-10-17 at 16 21 12](https://github.com/user-attachments/assets/42a54ebb-981e-4a17-b912-367871955973)

One thing we found that makes it not hang is settings CPU limits for the container, which at first made no sense and that's why we started digging further. But when unbound we are hanging on 10-30% of the spawned containers at different random points of the code. Later we found it to be erlang, and reverted to 26.2.5.3.

**Comment by garazdawi** [comment]
Are things working fine with OTP-26.2.5.3? Would you be able to do a bisect to figure out which specific commit it is that introduces the issue?

**Comment by rafaelgaspar** [comment]
> Are things working fine with OTP-26.2.5.3? Would you be able to do a bisect to figure out which specific commit it is that introduces the issue?

Unfortunately I don't have a reliable way to bisect this as this, as it is happening on the containers in the servers and we could not reproduce in the development environment.

**Comment by rafaelgaspar** [comment]
One unconfirmed theory of mine is that it might have come from [this PR](https://github.com/erlang/otp/pull/8892), since it was fixing a similar issue, but it's just a guess at the moment.

**Comment by rafaelgaspar** [comment]
> Are things working fine with OTP-26.2.5.3?

We have now confirmed that OTP-26.2.5.3 works as expected, and are now in the process of reverting all services back to it.

**Comment by RaimoNiskanen** [comment]
The process tree you are showing is normal for any Erlang VM that has done a resolver lookup through the default resolver backend `inet_gethost_native`.  There is one supervising inet_gethost program that starts up to 4 worker children.  They are used to call the OS resolver in a thread safe manner.

I don't think this says anything about where in Erlang code your application is hanging.

**Comment by seriyps** [comment]
I think `epgsql` CI fell victim of this issue. As you see, it succeeds on each OTP version from OTP-21 to OTP-27, but fails with timeout on OTP-26.2.5.4
https://github.com/epgsql/epgsql/actions/runs/11610405553

**Comment by seriyps** [comment]
I tried to reproduce it on 26.2.5.3 and it does not reproduce. So only affects 26.2.5.4.

Stes to reroduce:

```
# while on OTP-26.2.5.4
git clone https://github.com/epgsql/epgsql.git
cd epgsql
make common-test
```

**Comment by IngelaAndin** [comment]
Is it reproducible on OTP-26.2.5.5 ? 

Includes this fix that could be relevant if you are using TLS or crypto in some other way. So I do not know, just saying as this seems to be a 26.2.5.4 issue only and this is one of the fixes in 26.2.5.5 released this morning.

```
OTP-19316    Application(s): crypto
               Related Id(s): GH-8971, PR-8979

               Fixed bug that could cause blocking scheduler threads
               in crypto:supports(curves) if called the first time by
               more than one Erlang process. Bug exists only in
               OTP-26.2.5.4.

```

**Comment by seriyps** [comment]
What is interesting is that if I add ` --sname epgtest --setcookie epgtest` to the `rebar3 ct` command and then at the same time start the remote shell `r 'epgtest@laptop'` then I'm only able to connect at 1st few seconds. After CT blocks, I no longer able to remote-connect.
Also, I tried to do `[erlang:process_info(P) || P <- processes()].` in this remote shell and it also locked and later remote shell was kicked out due to lack of pings I guess

```
*** ERROR: Shell process terminated! (^G to start new job) ***
=ERROR REPORT==== 1-Nov-2024::14:45:20.797816 ===
** Node 'epgtest@seriy-laptop' not responding **
** Removing (timedout) connection **
```

**Comment by seriyps** [comment]
@IngelaAndin will check now

**Comment by seriyps** [comment]
@IngelaAndin yes, issue no longer reproduce on OTP-26.2.5.5. Guess #8971 fixed it. Thanks!

**Comment by seriyps** [comment]
Re-running the `epgsql` CI on 26.2.2.5 also succeeded https://github.com/epgsql/epgsql/actions/runs/11610405553 :+1:

**Comment by garazdawi** [comment]
Fixed by #8971
